assets/js/blocks/stock-filter/block.js

```diff
@@ -48,10 +34,31 @@ const StockStatusFilterBlock = ( {
 	attributes: blockAttributes,
 	isEditor = false,
 } ) => {
+	const hideOutOfStockItems = useMemo(
+		() => getSetting( 'hideOutOfStockItems', false ),
+		[]
+	);
+	const { outofstock, ...otherStockStatusOptions } = useMemo(
+		() => getSetting( 'stockStatusOptions', {} ),
+		[]
+	);
+	const STOCK_STATUS_OPTIONS = useMemo( () => {
```

Comment:
Again, seems like an over-optimisation personally for the same reasons as above.